File: computer_science/99problems/zig/1.01.zig
const std = @import("std");

pub fn ttyFindLast(list: [4]u8) void {
    const length = list.len;

    if (length == 0) {
        std.debug.print("The list is empty.\n", .{});
    } else {
        var last: usize = length - 1;
        std.debug.print("{c}\n", .{ list[last] });
    }
}

pub fn main() void {
    const list = [4]u8 { 'a', 'b', 'c', 'd' };

    ttyFindLast(list);
}
// => d

File: computer_science/99problems/lua/1.01.lua
local list = {"a", "b", "c", "d"}

function tty_find_last(l)
    local length = #l

    if length == 0 then
        print("This list is empty")
    else
        print(l[length])
    end
end

tty_find_last(list)
-- => d
